### PR TITLE
Significantly speed-up make

### PR DESCRIPTION
**Pull request by sttts**
- do not call `dirname` for every go file => gives **>8 seconds** on each make run
- do not preserve time during rsync from `_output/local/go/bin` to `_output/local/bin/<os>/<arch>`:
  
  We do a touch after our rebuild heuristic triggers.
  
  If in `_output/local/go/bin` you have an older binary and go decides that it's good enough, the following rsync will "preserve the timestamp" of the binary copy in `_output/local/bin/<os>/<arch>`. This reverts the effect of any previous `touch` (compare `Makefile.generated_files`).

  We do a complete rsync of binaries for each go target, not only the one which is rebuilt. This means that later go targets will overwrite the timestamps of freshly touched binaries. Hence, our rebuild heuristic will trigger again and again when running `make`.

  This PR remove the "preserve mtime" from the rsync call. Hence, the effect of touch is not reverted anymore by later rsyncs.

### CODE REVIEW

**Comment by sttts** [comment]
Another opportunity for optimisation: we call `kube::golang::build_kube_toolchain` multiple times during the build.

File: hack/make-rules/helpers/cache_go_dirs.sh [hunk 1]
```diff
@@ -60,9 +60,9 @@ fi
 mkdir -p $(dirname "${CACHE}")
 if $("${NEED_FIND}"); then
     kfind -type f -name \*.go  \
-        | xargs -n1 dirname    \
-        | LC_ALL=C sort -u     \
+        | sed 's|/[^/]*$||'    \
```
Comment (thockin) [inline hack/make-rules/helpers/cache_go_dirs.sh]: worth also changing to `sed -e 's|/[^/]*$||' -e 's|^./||'` ?

**Comment by thockin** [comment]
Welcome to the dark side.  The toolchain one has been on my list but I have just not had time.

File: hack/make-rules/helpers/cache_go_dirs.sh (hunk 1 above)
Comment (sttts) [inline hack/make-rules/helpers/cache_go_dirs.sh]: I don't expect much improvement. The issue was that dirname was called once per file. Sed only runs once.

**Pull request merged**